### DISCUSSION TITLE
"binwalk -E" and huge files (multiple Giga Bytes)

### DISCUSSION THREAD

**Discussion by ricardomenzer** [OSPG/binwalk#18] [Q&A] [answered]
Hello! I have a question about using `binwalk` to calculate the entropy in a huge file.
I'm trying to recover files from a SSD that was `fdisk`'ed and `mkfs`'ed (and may be more, it was an automatic system that "wiped" the SSD, so I'm not exactly sure what operations were done).

The SSD before the incident was quite full of H.264 video files.
`testdisk`and `photorec` didn't return anything useful. 

My next move is to run `binwalk -E` on that disk and analyse the entropy on it to see if I can still have some hope.
My though is if the entropy is high (or have some spikes somewhere in the disk), then there may be data there. On the other hand, if the entropy is mostly low, then my chances of getting something are low too.
So, I'm trying to determine if it is worth to spend more time trying to recover the files or I should just accept the loss and move on.

Ok, enough sad story. The actual question is: Is `binwalk -E` capable of handling a huge file of several GB? I'm talking of a 512GB SSD. Or it is better to find another tool for that?

**Comment by stkw0** [comment, marked as answer]
It is capable of handling huge file sizes, but it would be very slow. I am unaware if there are other tools optimized for this task.

**Reply by ricardomenzer** [reply]
Thanks. I used it and it worked fine.
